src/testing/core/__tests__/createTestSchema.test.tsx

```diff
@@ -1020,12 +1019,217 @@ describe("schema proxy", () => {
+            }),
+          },
+        },
+      });
+
+      using _fetch = createSchemaFetch(resetTestSchema);
+
+      const client = new ApolloClient({
+        cache: new InMemoryCache(),
+        uri,
+      });
+
+      const query: TypedDocumentNode<ViewerQueryData> = gql`
+        query {
+          viewer {
+            id
+            name
+            age
+            book {
+              id
+              title
+              publishedAt
+              ... on ColoringBook {
+                colors
+              }
+            }
+          }
+        }
+      `;
+
+      const Fallback = () => {
+        useTrackRenders();
+        return <div>Loading...</div>;
+      };
+
+      const App = () => {
+        return (
+          <React.Suspense fallback={<Fallback />}>
+            <Child />
+          </React.Suspense>
+        );
+      };
+
+      const Child = () => {
+        const result = useSuspenseQuery(query);
+
+        useTrackRenders();
+
+        Profiler.mergeSnapshot({
+          result,
+        } as Partial<{}>);
+
+        return <div>Hello</div>;
+      };
+
+      const { unmount } = renderWithClient(<App />, {
+        client,
+        wrapper: Profiler,
+      });
+
+      // initial suspended render
+      await Profiler.takeRender();
+
+      {
+        const { snapshot } = await Profiler.takeRender();
+
+        expect(snapshot.result?.data).toEqual({
+          viewer: {
+            __typename: "User",
+            age: 42,
+            id: "1",
+            name: "String",
+            book: {
+              __typename: "TextBook",
+              id: "1",
+              publishedAt: "2024-01-01",
+              // value set in this test with .add
+              title: "The Waves",
+            },
+          },
+        });
+      }
+
+      unmount();
+    });
+    it("resets the schema to the original", async () => {
+      const Profiler = createDefaultProfiler<ViewerQueryData>();
+
+      resetTestSchema.reset();
+
+      using _fetch = createSchemaFetch(resetTestSchema);
+
+      const client = new ApolloClient({
+        cache: new InMemoryCache(),
+        uri,
+      });
+
+      const query: TypedDocumentNode<ViewerQueryData> = gql`
+        query {
+          viewer {
+            id
+            name
+            age
+            book {
+              id
+              title
+              publishedAt
+              ... on ColoringBook {
+                colors
+              }
+            }
+          }
+        }
+      `;
+
+      const Fallback = () => {
+        useTrackRenders();
+        return <div>Loading...</div>;
+      };
+
+      const App = () => {
+        return (
+          <React.Suspense fallback={<Fallback />}>
+            <Child />
+          </React.Suspense>
+        );
+      };
+
+      const Child = () => {
+        const result = useSuspenseQuery(query);
+
+        useTrackRenders();
+
+        Profiler.mergeSnapshot({
+          result,
+        } as Partial<{}>);
```

Comment:
```suggestion
      const Fallback = () => {
        return <div>Loading...</div>;
      };

      const App = () => {
        return (
          <React.Suspense fallback={<Fallback />}>
            <Child />
          </React.Suspense>
        );
      };

      const Child = () => {
        const result = useSuspenseQuery(query);

        Profiler.mergeSnapshot({
          result,
        });
```

Same here. No need for these 🙂

Reply:
Ah, yep thanks for calling this out :) Fixed in https://github.com/apollographql/apollo-client/pull/11760/commits/178eb83407b96d3d178193777dffebbe6ea2b1d4